packages/block-library/src/post-date/edit.js

```diff
@@ -124,10 +126,15 @@ export default function PostDateEdit( {
 									<PublishDateTimePicker
 										currentDate={ date }
 										onChange={ setDate }
-										is12Hour={ is12HourFormat(
-											siteTimeFormat
-										) }
+										is12Hour={ is12Hour }
 										onClose={ onClose }
+										dateOrder={
+											is12Hour
+												? /* translators: Order of day, month, and year when the 12-hour clock is enabled. Available formats are 'dmy', 'mdy', and 'ymd'. */
+												  _x( 'mdy', 'date order' )
+												: /* translators: Order of day, month, and year when the 12-hour clock is disabled. Available formats are 'dmy', 'mdy', and 'ymd'. */
+												  _x( 'dmy', 'date order' )
+										}
```

Comment:
Unfortunately, for component back compat I think we need to maintain current behavior at the component level, not at the app level. So the default `dateOrder` for DateTimePicker/TimePicker itself is going to remain `is12Hour ? 'mdy' : 'dmy'`. (Which will of course be overridden when an explicit `dateOrder` is passed in by the consumer.)

Reply:
Also, I think the original logic was meant to be a heuristic that is basically "Do they use 12-hour time? Then it's probably US/Canada, so use MDY" 😅 Which is fine for a crude heuristic, but now that we can specify proper date order based on locale translations, I think it should be based solely on the translation (without the 12-hour logic):

```suggestion
										dateOrder={
											/* translators: Order of day, month, and year. Available formats are 'dmy', 'mdy', and 'ymd'. */
											 _x( 'dmy', 'date order' )
										}
```